test/parallel/test-timers-unref-reuse-no-exposed-list.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+
+const timer1 = setTimeout(() => {}, 1).unref();
+assert.strictEqual(timer1._handle._list, undefined,
+                   'timer1._handle._list should be undefined');
+
+// Check that everything works even if the handle was not re-used.
+setTimeout(() => {}, 1);
+const timer2 = setTimeout(() => {}, 1).unref();
+// Note: It is unlikely that this assertion will be hit in a failure.
```

Comment:
That comment seems to be specific to a previous version of this PR where it would throw in that case. I don't think it's relevant in the final version.